File: Interviews/LeetCode/Daily-Problems/2022/October/Day-11/334.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        ans, n = [], len(nums)
        for i in range(n):
            if len(ans) == 0 or nums[i] > ans[-1]:
                ans.append(nums[i])
            else:
                idx = bisect.bisect_left(ans, nums[i])
                ans[idx] = nums[i]
            if len(ans) > 2:
                return True
            
        return False

File: Interviews/LeetCode/Topic-Wise/Binary-Search/Medium/Increasing-Triplet-Subsequence/334.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++) {
            if (ans.size() == 0 or nums[i] > ans.back()) {
                ans.push_back(nums[i]);
            } else {
                int lb = lower_bound(ans.begin(), ans.end(), nums[i]) - ans.begin();
                ans[lb] = nums[i];
            }

            if (ans.size() > 2) return true;
        }

        return false;
    }
};